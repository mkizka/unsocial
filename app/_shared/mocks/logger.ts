import { mockDeep } from "jest-mock-extended";

import { createLogger } from "@/_shared/utils/logger";

const mockedLogger = mockDeep<ReturnType<typeof createLogger>>();

jest.mock("@/_shared/utils/logger");
const mockedCreateLogger = jest.mocked(createLogger);
mockedCreateLogger.mockReturnValue(mockedLogger);

export { mockedLogger };
