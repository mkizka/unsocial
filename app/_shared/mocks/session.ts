import { userSessionService } from "@/_shared/user/services/userSessionService";

jest.mock("@/_shared/user/services/userSessionService");
export const mockedGetSessionUserId = jest.mocked(
  userSessionService.getSessionUserId,
);
// export const mockedGetSessionUser = jest.mocked(getSessionUser);
