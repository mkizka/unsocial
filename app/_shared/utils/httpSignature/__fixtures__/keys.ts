const publickKey = `-----BEGIN PUBLIC KEY-----
MIIBHjANBgkqhkiG9w0BAQEFAAOCAQsAMIIBBgKB/gDJ5rcs3GJ+7YNOZ2qYgBMX
77suMZmpXNo1YQHyi4ZQkbvojsv7fuDUwjSR7CL78xN88pnTgJTrJfMjeH3LCV/R
BvcsRQNK5QRLvD9ieGLo+1cNycyr1WScTz3RmfkCAGhe4vhmBIlne9w1ItwrFAnP
qmFuAswgEXH24WMLGocPo/fskUoiqtPzqw73r2FvXHv9Qv588lswy3NsL7iANExz
fYzWy8nsa05UH1SrEJ9uT+Bh6ffupJtPuZ/lDSMl2qmHs4kQX67V1amKHl6U6eyx
/CLySZ7hI/SZfXE4ppyL/l+se8ti1I0OcUVMSXjyIN1Z2SNHPHu2QGx2/EGRAgMB
AAE=
-----END PUBLIC KEY-----
`;

const privateKey = `-----BEGIN PRIVATE KEY-----
MIIEqAIBADANBgkqhkiG9w0BAQEFAASCBJIwggSOAgEAAoH+AMnmtyzcYn7tg05n
apiAExfvuy4xmalc2jVhAfKLhlCRu+iOy/t+4NTCNJHsIvvzE3zymdOAlOsl8yN4
fcsJX9EG9yxFA0rlBEu8P2J4Yuj7Vw3JzKvVZJxPPdGZ+QIAaF7i+GYEiWd73DUi
3CsUCc+qYW4CzCARcfbhYwsahw+j9+yRSiKq0/OrDvevYW9ce/1C/nzyWzDLc2wv
uIA0THN9jNbLyexrTlQfVKsQn25P4GHp9+6km0+5n+UNIyXaqYeziRBfrtXVqYoe
XpTp7LH8IvJJnuEj9Jl9cTimnIv+X6x7y2LUjQ5xRUxJePIg3VnZI0c8e7ZAbHb8
QZECAwEAAQKB/WbVoE61IWXMy6hPfDeYZCE49XmdwcpFhM7TTg6fihXXZIIa2UKL
aTB0ihoYz+jL8g0d88OKJk9u0NwsLsYPTEJG7fYaYIbxBrMcepxI/Pf1rG1Vb23G
Dqyen7n2VqHyWzaZHDO589JFf++AhetbFXQ0L9O8kQN0XzuDqNKq/Dk69LwBlnxX
gK5BROV03coTwwBZGkX+WaSBWmRxurWaN4Mskpsv0ITXi8CD9dyIvSoFlig+UV1Q
2YqLr9O7QUkEnEH2OwaBowilOfEgIokY5f0TseuGdFQ1Pue+l39wiq5mWUjDWpzY
ja2wc2JeL70ph8HRZFPPszclYtTA1RECfw7DL5tIJUUdjIUkuEP0jlMeojTgW5bm
aZzw1nv694ftgLqFclkIfvU6KVIOdVfSWCcCedFJnIm31KmBrQJyCUV9cYg/Mdge
/EiWWATjl+FkIzkHOVw35bpBjQKmTV45Nmf9zzuIkdRKtRqettBimLVRfShqmbXh
h2xpO19aqYcCfw2tOt+j3hv+vt/sQR19oTT0leydNZpsRbvFwtdjLLZ3mCym04l+
XR8SVxps+jHJDGNZyvT3ps0eLL9FtF6rwDxpU46LN9y1EwtT+ZT7TtYpTX3KbrMt
yzsegf6cACErFvDLV4rGeMfZfhUHhxG3btwywymJ+hT9hPhjbp0zoicCfwL/+z6t
DJXdCV9NNPzqR7nwcboZFcO2kyPHdW7Ac2j2BMDiuXgKV/JxnVAFjCcKoctbn3G+
acc6ykagCY6DqgqrEVVEslZfBZmio8Rqk8TD2ryOEUQCP/b3tzYgoXxSWBIN6/0H
JctkWyVLt47lWm8OXkbGUvI5IJIRWHVqMSUCfwEVe5cB27aISARjKv31WPNjdwZb
p0Xn0tcA42JCn+aOL+dKCXXaJ/MwdurHPET51BuDoEwt3HBhNdaq5ZKjNgcTqeWg
Lvs09p9tNliSWjsq5tNTWdaliHfVASR2pKP8UEMhOnzrIeNk31M3V4MV8JlPBJyk
LMfBF+q/pqKa1v0Cfw3+go4Vmz0JyVex74CpdqNXrjvHU873ypjSOemeoRP5T8qz
Pw9+4NLFLb7E9vZTCLGyBW4OCKb0fyrbkWDAh/9/gv8hgvtM71Fgj8iFR0FrgG4b
s+7Y1gr9CevUwUnFFtagFM2oEnOYydaqxq5BvXgFdoH6Whr9Gl03RQOl7QU=
-----END PRIVATE KEY-----
`;

export const mockedKeys = { publickKey, privateKey };
