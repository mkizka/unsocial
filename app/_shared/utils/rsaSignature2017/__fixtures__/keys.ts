export const rsaKeyPair = {
  private:
    "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEAxiD/F5M/LPKsNeTV1R55dtsQRYqP0835BuxuYyAeTENDwpji\neFZHztrmpPFqnfPgGlW3aDfN84PfC5F+LznB1EmWlfQwWTTTb7VurlMmHOgQx9c+\n2+Yg22cgPGh8qTxUALZFpQ+DHVBU59FXbzpZKeF2go9fqvJvYaxcf5Sq1pu/Y62k\nplCVxmow508agslRvqAzowsHvB/Dp5wETaXiyvXeqCFSEmTDCu3CCnu4HuwwP+H3\nKNSuAPZtcjOlCxrvO50RvzT5RnnRITuTi4oimIJMQoIX6N/RZzgFRofsTBKqbjFs\nZ6hja496yLp/3KJ8egQ2srl4kt3TzHlDfKXPnQIDAQABAoIBAF1mirrjaGHZGMPj\nV8a7MgmM5dPSqVDxBZd09+j06DONdjETf5e5HOrwGNDjGh7SK/BePjL8M7O7k/dR\nJEpBERXxhvK0BeLX22ELB91YR1xTLi33U6O4jf7AU9sGatYhIXygcJ6KfIXiK9SI\nxI4j1IrAxexwiGn/FfDWn1hpjkvSlUQr14YCCuckBXAkAAjqlcrH0xEkcFls4HAo\nwn+vCaxN+tQaP9fmwbBMyILSLgmirFeZPJbtL731h677jED0Gre2PQJc2UzNnWA2\nThKlIqikTbyNAULC9u3+dtSqARD2NWTh8rhnkyiPQ3giYezWnbDX7/gEZx6ZhjmA\nA5Bc0kUCgYEA9P4khD7fBQURD84boR6WsCd6VEpBuDqEzx6HDIUTwkj2VG6ftzz6\nTwCZmhgnRCedutCzKqfH4TCMlXUYoXgg/8lrmwVxWCYt41NWr4cMS3GThXwEBrDN\nHUu+BCEEWmUdaILXlpD7XFz2bOi45cj8jp/UQSZKC9wKIJwT2OLuydMCgYEAzwfV\nzxfxGRf/MfN6nhEGVgJ03YpspIgIkLZQctZOBTPBUXsECYj85cAdPtPwXbHHqx4u\nW63FqHhOBZdf8b8a8zOCTjnxiZAzlTGezeK9jMtQM5+IgEWNf46Zzt+fLnooq4Rc\nGcgsNwp3xuBMKPPiCEXggUlICAEoEAB1e5E8Ks8CgYB6oBZhByMT3SRcF9rme2wL\nQcdOF1jrwIanMO2By75HGcQKINDao6/aCT/YzEwjJf0h+8IiEJbQVT+Ys2Kft4Q0\n43lzv3arG3kOdrFF7D+wJ2oifx94xdf02OuEH46RPFlK7WrBV1IpOfKyMo4Rndck\nAEX6UhLv+rQq/FRG09eXtwKBgQC0MrqsFce5ssRh4u4gNOtR47N6vSiREQJ7iuR9\nAAFKz+B9HljGP+D3zP7nl6uIsdv8Gjn5fEglugkXHsnY7iDz/Cp4EkElHkl3m4gA\n5lFcAf8d38b89w0RMLq5cLTpzFQn7f4IRs4NXmQgy0we/SAJWWDuxy0WIoOPYgbf\n9errywKBgQDC7NPm+2k8Mw05JRKQ+yQRhsEOh3/gDFz28w/DPqmm5Z5YOhVIK5bd\n1mSEklsX8dSt5YObMcVavt6+gWi3lPkznO063BzpnPDCBTbApyaB7B8hlKy/KtbZ\nzwYQC2O+AflQIvjZu3NaAAZ288V9YVz6pys9ycBweOoQuJkovLIsCg==\n-----END RSA PRIVATE KEY-----\n",
  public:
    "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxiD/F5M/LPKsNeTV1R55\ndtsQRYqP0835BuxuYyAeTENDwpjieFZHztrmpPFqnfPgGlW3aDfN84PfC5F+LznB\n1EmWlfQwWTTTb7VurlMmHOgQx9c+2+Yg22cgPGh8qTxUALZFpQ+DHVBU59FXbzpZ\nKeF2go9fqvJvYaxcf5Sq1pu/Y62kplCVxmow508agslRvqAzowsHvB/Dp5wETaXi\nyvXeqCFSEmTDCu3CCnu4HuwwP+H3KNSuAPZtcjOlCxrvO50RvzT5RnnRITuTi4oi\nmIJMQoIX6N/RZzgFRofsTBKqbjFsZ6hja496yLp/3KJ8egQ2srl4kt3TzHlDfKXP\nnQIDAQAB\n-----END PUBLIC KEY-----\n",
};
