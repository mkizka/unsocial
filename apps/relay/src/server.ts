import Fastify from "fastify";

import { relayActivity } from "./relay";

export const startServer = () => {
  const fastify = Fastify({
    logger: {
      transport: {
        target: "@fastify/one-line-logger",
      },
    },
  });

  fastify.post<{
    Body: {
      id?: string;
    };
  }>("/relay", (request, reply) => {
    if (!request.body.id) {
      return reply.code(400).send();
    }
    relayActivity(request.body.id);
    reply.code(202).send();
  });

  const port = Number(process.env.PORT) || 3001;
  fastify.listen({ port });
};
