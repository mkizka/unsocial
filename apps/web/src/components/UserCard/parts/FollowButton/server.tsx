import { prisma } from "@soshal/database";

import { getServerSession } from "@/utils/getServerSession";

import { FollowButton as Client } from "./client";

export async function FollowButton({ followeeId }: { followeeId: string }) {
  const session = await getServerSession();
  const sessionUserId = session?.user?.id;
  if (!sessionUserId) {
    return null;
  }
  const follow = await prisma.follow.findFirst({
    where: {
      followeeId,
      followerId: sessionUserId,
    },
  });
  return <Client followeeId={followeeId} followStatus={follow?.status} />;
}
