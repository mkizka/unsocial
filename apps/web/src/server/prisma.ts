// Stryker disable all
import { PrismaClient } from "@soshal/database";

import { env } from "@/utils/env";

export const prisma = new PrismaClient({
  log: env.NODE_ENV === "development" ? ["query", "error", "warn"] : ["error"],
});
