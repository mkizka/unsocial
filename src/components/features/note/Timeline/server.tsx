import { noteService } from "@/server/service";

import { ClientComponent } from "./client";

type Props = {
  userId?: string;
};

export async function Timeline({ userId }: Props) {
  const notes = await noteService.findManyNoteCards({ userId, count: 10 });
  return <ClientComponent firstLoadedNotes={notes} />;
}
