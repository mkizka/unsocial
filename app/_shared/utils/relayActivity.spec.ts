import type { Credential } from "@prisma/client";
import type { AP } from "activitypub-core-types";
import { captor } from "jest-mock-extended";
import { http, HttpResponse } from "msw";

import { mockedKeys } from "@/_shared/mocks/keys";
import { mockedPrisma } from "@/_shared/mocks/prisma";
import { server } from "@/_shared/mocks/server";

import {
  relayActivityToFollowers,
  relayActivityToInboxUrl,
} from "./relayActivity";

jest.useFakeTimers();
jest.setSystemTime(new Date("2020-01-01T00:00:00Z"));

jest.mock("@/../package.json", () => ({
  version: "1.2.3",
}));

describe("relayActivity", () => {
  describe("relayActivityToInboxUrl", () => {
    test("正常系", async () => {
      // arrange
      const dummyUserId = "dummy_userId";
      mockedPrisma.credential.findUnique.mockResolvedValueOnce({
        privateKey: mockedKeys.privateKey,
      } as Credential);
      const headerFn = jest.fn();
      const headerCaptor = captor();
      const bodyFn = jest.fn();
      const bodyCaptor = captor();
      server.use(
        http.post("https://remote.example.com/inbox", async ({ request }) => {
          headerFn(Object.fromEntries(request.headers));
          bodyFn(await request.json());
          return new HttpResponse(null, { status: 202 });
        }),
      );
      // act
      await relayActivityToInboxUrl({
        userId: dummyUserId,
        inboxUrl: new URL("https://remote.example.com/inbox"),
        activity: {
          type: "Dummy",
          actor: "https://myhost.example.com/users/dummy_other_userId/activity",
        } as unknown as AP.Activity,
      });
      // assert
      expect(headerFn).toHaveBeenCalledWith(headerCaptor);
      expect(headerCaptor.value).toMatchSnapshot();
      expect(bodyFn).toHaveBeenCalledWith(bodyCaptor);
      expect(bodyCaptor.value).toEqual({
        type: "Dummy",
        actor: "https://myhost.example.com/users/dummy_other_userId/activity",
      });
    });
  });

  describe("relayActivityToFollowers", () => {
    test("正常系", async () => {
      // arrange
      const dummyUserId = "dummy_userId";
      mockedPrisma.follow.findMany.mockResolvedValue([
        {
          // @ts-ignore
          follower: { inboxUrl: "https://remote1.example.com/users/foo/inbox" },
        },
        // @ts-ignore
        { follower: { inboxUrl: "https://remote2.example.com/inbox" } },
        // @ts-ignore
        { follower: { inboxUrl: "https://remote2.example.com/inbox" } },
        // @ts-ignore
        { follower: { inboxUrl: null } },
      ]);
      mockedPrisma.credential.findUnique.mockResolvedValueOnce({
        privateKey: mockedKeys.privateKey,
      } as Credential);
      const headerFn = jest.fn();
      const headerCaptors = [captor(), captor()];
      const bodyFn = jest.fn();
      const bodyCaptors = [captor(), captor()];
      const inbox1 = http.post(
        "https://remote1.example.com/users/foo/inbox",
        async ({ request }) => {
          headerFn(Object.fromEntries(request.headers));
          bodyFn(await request.json());
          return new HttpResponse(null, { status: 202 });
        },
      );
      const inbox2 = http.post(
        "https://remote2.example.com/inbox",
        async ({ request }) => {
          headerFn(Object.fromEntries(request.headers));
          bodyFn(await request.json());
          return new HttpResponse(null, { status: 202 });
        },
      );
      server.use(inbox1, inbox2);
      // act
      await relayActivityToFollowers({
        userId: dummyUserId,
        // @ts-ignore
        activity: { type: "Dummy" },
      });
      // assert
      expect(headerFn).toHaveBeenNthCalledWith(1, headerCaptors[0]);
      expect(headerCaptors[0]!.value).toMatchSnapshot();
      expect(headerFn).toHaveBeenNthCalledWith(2, headerCaptors[1]);
      expect(headerCaptors[1]!.value).toMatchSnapshot();
      expect(bodyFn).toHaveBeenCalledWith(bodyCaptors[0]);
      expect(bodyCaptors[0]!.value).toEqual({ type: "Dummy" });
      expect(bodyFn).toHaveBeenCalledWith(bodyCaptors[1]);
      expect(bodyCaptors[1]!.value).toEqual({ type: "Dummy" });
    });
  });
});
