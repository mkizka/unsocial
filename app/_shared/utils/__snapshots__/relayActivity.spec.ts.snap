// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`relayActivity relayActivityToFollowers 正常系 1`] = `
{
  "accept": "application/activity+json",
  "content-type": "application/activity+json",
  "date": "Wed, 01 Jan 2020 00:00:00 GMT",
  "digest": "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  "host": "remote1.example.com",
  "signature": "keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="XXDdO7dhc59HNCo3zrehwGu2wFPRHRIqfl7rrobYaU5rMX0nqKRAF4wjxFk7YiEx0oA+5WoP5QlqVDzqNY32WwtRjh0NRwWzHXdCaZSTFE4Icsu+eLParFkFgzdx9vV50bZ1L8vxv++eEwChos/p5O6o/JS9SkGzJVpfIt2/hZBJN7kFIbXW4vYzO9uiVKVPjf8MgaKgOkl8oVmP9rVGrwalUz4plxwYCfBUHt1QZvkIS/M+w4gT7/QYh63J5bmoP9aREnlMxggH07cWjt3KTMFpfBCk5RuPTNdtAz5kRXYtWWznxxCucRBYWTE9yHFT4aRc7tSehc96vX32qg=="",
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
}
`;

exports[`relayActivity relayActivityToFollowers 正常系 2`] = `
{
  "accept": "application/activity+json",
  "content-type": "application/activity+json",
  "date": "Wed, 01 Jan 2020 00:00:00 GMT",
  "digest": "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  "host": "remote2.example.com",
  "signature": "keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="jvXb6A4uqZe4EKeU5a825JA/YX5VdyE3cQpvmwPxmGncx7N3KMfeLNGCNK9ikfhCDzixbugdFasUom8DcjqJQqNrDj9RUkqjZwfeec3t3sRp9ZiBa8TY++KWyZOBc0oWpgReUvWYQo1+JgAkKQcdERkQpQ2pONJU/CF17qMBMjkHXFKknkHJqenXf9XN5aLug1gPKq/HFYciwWXCbPNvCuucIUyRL06Le6LvEBZSlGbHO/myyPpXzwyhglo8OX8RPlr5vteQQEMDDDAshG+00ybQlBu11ozp8wptr1FVrdylRQ2RtNjbWesUcrYuey8JgwY+aCN86iZA4NLaRw=="",
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
}
`;

exports[`relayActivity relayActivityToInboxUrl 正常系 1`] = `
{
  "accept": "application/activity+json",
  "content-type": "application/activity+json",
  "date": "Wed, 01 Jan 2020 00:00:00 GMT",
  "digest": "SHA-256=Gwtaf2C1j/hnbwtC6f37u2T6SiitMcs15298sjIcR5w=",
  "host": "remote.example.com",
  "signature": "keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="jKd5MWWaxNkuXfLrV+hnhs2xOJ0DOI7rcQFjOa9eTKw0UI83AKLn+AfMAEWGZjO97jYP5OvTk2gjOeJZ2imo3E4KnfrfUH1MG2GwhJQjhpjeAu6oluzZZ1Ct8Qjz1kf25ZvlMAO8qbpoMcvwedOAw9wpDgZagddm/y/xCMCaCOYb3P4n4Zinv/k+EcAnVWjPFm+i3optmghZ9gQST8D4LG3Trzv/JsWhhwFTs9BhVH8MaXM33ELn1KPw2ECX7bx4YMG105uaRmLDM1tmxMDPjQZb43fX0TQEbtK0ohQ56upOpGCds/B2F0iGsx+zMsUzObxQVS3E9HPAC76xpQ=="",
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
}
`;
